### PR TITLE
feat: merge tooltip feature to master (SDKCF-5948)

### PR DESCRIPTION
**Pull request by maureenorea-clores**
# Description
Merges tooltip feature from `feature/tooltip-poc`, with some modifications due to major refactor since v7.2.0.

# Checklist
- [x] I have read the [contributing guidelines](../CONTRIBUTING.md).
- [x] I wrote/updated tests for new/changed code
- [x] I removed all sensitive data **before every commit**, including API endpoints and keys
- [ ] I checked that all possible internal and platform/library exceptions are gracefully handled and **will not** crash the host app
- [ ] I checked for open & known issues (especially crashes) for any newly added platform APIs or libraries
- [x] I tested non-trivial changes on a real device
- [x] I added ticket/changes in changelog
- [x] I ran `./gradlew check` without errors

### CODE REVIEW

File: inappmessaging/src/main/java/com/rakuten/tech/mobile/inappmessaging/runtime/manager/DisplayManager.kt [hunk 1]
```diff
@@ -33,23 +43,150 @@ internal interface DisplayManager {
...
+                        activity = activity
+                    )
+                }
+            }
+        }
+
+        private fun scheduleRemoval(delay: Int, view: ViewGroup, id: String? = null, activity: Activity) {
+            if (delay > 0) {
+                Handler(Looper.getMainLooper()).postDelayed(
+                    { removeCampaign(view, id, activity) }, delay * MS_MULTIPLIER
```
Comment (vincentdelmundo) [inline inappmessaging/src/main/java/com/rakuten/tech/mobile/inappmessaging/runtime/manager/DisplayManager.kt:112]: How did you handle repo and impression update when auto-disappear?

File: inappmessaging/src/main/java/com/rakuten/tech/mobile/inappmessaging/runtime/workmanager/workers/DisplayMessageWorker.kt [hunk 2]
```diff
@@ -47,15 +46,17 @@ internal class DisplayMessageWorker(
...
+        val messages = messageReadinessManager.getNextDisplayMessage()
         val hostActivity = InAppMessaging.instance().getRegisteredActivity()
-        val imageUrl = message.getMessagePayload().resource.imageUrl
         if (hostActivity != null) {
-            if (!imageUrl.isNullOrEmpty()) {
-                fetchImageThenDisplayMessage(message, hostActivity, imageUrl)
-            } else {
-                // If no image, just display the message.
-                displayMessage(message, hostActivity)
+            for (message in messages) {
```
Comment (vincentdelmundo) [inline inappmessaging/src/main/java/com/rakuten/tech/mobile/inappmessaging/runtime/workmanager/workers/DisplayMessageWorker.kt:52]: Please confirm if we need to handle normal campaigns on top of tooltip campaigns

**Review by vincentdelmundo — CHANGES_REQUESTED** [review]
Looks good. Just a few comments. Please check.
Also, try to increase code coverage if possible.

File: inappmessaging/src/main/java/com/rakuten/tech/mobile/inappmessaging/runtime/manager/DisplayManager.kt (hunk 1 above)
Comment (maureenorea-clores) [reply inappmessaging/src/main/java/com/rakuten/tech/mobile/inappmessaging/runtime/manager/DisplayManager.kt]: Sorry it was't handled yet. Added handling

File: inappmessaging/src/main/java/com/rakuten/tech/mobile/inappmessaging/runtime/workmanager/workers/DisplayMessageWorker.kt (hunk 2 above)
Comment (maureenorea-clores) [reply inappmessaging/src/main/java/com/rakuten/tech/mobile/inappmessaging/runtime/workmanager/workers/DisplayMessageWorker.kt:52]: Retained the current behavior of feat/tooltip-poc, where normal campaigns and tooltip campaigns don't overlap. Will create separate ticket if needed

**Comment by maureenorea-clores** [comment]
@vincentdelmundo I've added unit tests to increase coverage

**Review by vincentdelmundo — APPROVED** [review]
I think it is better to do the QA/testing first before merging it to master. Just to make sure that everything is working.
What do you think?

**Comment by maureenorea-clores** [comment]
@vincentdelmundo 
Yes I agree

**Pull request merged**